src/shogun/machine/gp/SingleLaplacianInferenceMethod.cpp

```diff
@@ -295,97 +492,54 @@ void CSingleLaplacianInferenceMethod::update_init()
 	m_Psi=Psi_New;
 }
 
-void CSingleLaplacianInferenceMethod::update_alpha()
+
+void CSingleLaplacianInferenceMethod::register_minimizer(Minimizer* minimizer)
 {
-	float64_t Psi_Old=CMath::INFTY;
-	float64_t Psi_New=m_Psi;
+        REQUIRE(minimizer, "Minimizer must set\n");
+	if (!dynamic_cast<SingleLaplacianNewtonOptimizer*>(minimizer))
+	{
+		FirstOrderMinimizer* opt= dynamic_cast<FirstOrderMinimizer*>(minimizer);
+		REQUIRE(opt, "The provided minimizer is not supported\n")
+	}
+        if(minimizer!=m_minimizer)
+        {
+                delete m_minimizer;
+                m_minimizer=minimizer;
+        }
+}
 
+void CSingleLaplacianInferenceMethod::update_alpha()
+{
+	SingleLaplacianNewtonOptimizer *opt=dynamic_cast<SingleLaplacianNewtonOptimizer*>(m_minimizer);
+	if (opt)
+	{
+		SG_REF(this);
+		opt->set_target(this);
+		opt->minimize();
+	}
+	else
+	{
+		FirstOrderMinimizer* minimizer= dynamic_cast<FirstOrderMinimizer*>(m_minimizer);
```

Comment:
@karlnapf 
Now, lbfgs and nlopt minimizers can be used in `SingleLaplacianInferenceMethod`.